code/src/components/TodoItem.js

```diff
@@ -0,0 +1,54 @@
+import React, { useEffect, useState } from "react";
+import { useDispatch, useSelector } from "react-redux";
+import { todos } from "../reducers/todos.js";
+import moment from 'moment'
+import { CustomCheckbox } from './CustomCheckbox';
+
+
+export const TodoItem = ({ itemIndex }) => {
+  const item = useSelector((store) => store.todos.list.items[itemIndex]);
+  const dispatch = useDispatch();
+
+  const handleCheckboxClick = (e) => {
+    dispatch(todos.actions.setDone({
+      itemIndex: itemIndex,
+      done: !item.done,
+    })
+    );
+  };
+
+  const onRemoveClicked = (e) => {
+    dispatch(
+      todos.actions.removeTodo({
+        itemIndex,
+      })
+    );
+  };
+
+
+  return (
+    <div className={`todo-item ${item.done ? "done" : ""}`}>
+      <CustomCheckbox
+        isChecked={item.done ? "done" : ""}
+        onChangeHandler={handleCheckboxClick}
+      />
+      <span className="todo-item-description"> {item.description} </span>
+      <div className="created-at-time"> {moment(item.date).fromNow()}</div>
+      <div className="todo-item-remove" onClick={onRemoveClicked}>
+        <svg className="trash"
```

Comment:
I would suggest to make this svg into its own icon component. It'll make it reusable and will make this TodoItem component a bit cleaner and easier to read.